client/src/pages/submitSecret/form.js

```diff
@@ -37,11 +39,27 @@ const Form = ({formResponse}) => {
     allow_insecure_passphrase: false
   }));
 
+  // HTML replacement
+  const INPUT_PASS_DEFAULT = "<input type=\"password\" id=\"passphrase\" name=\"passphrase\" class=\"form-control\">"
+  const INPUT_PASS_PUB = "<textarea rows=\"7\" cols=\"80\" id=\"passphrase\" name=\"passphrase\" placeholder=\"-----BEGIN PUBLIC KEY-----&#10;MIIBojANBgkqhkiG9w0BAQEFAAOCAY8AMIIBigKCAYEAvk3&#10;...&#10;-----END PUBLIC KEY-----\" onChange={handleInputChange} className=\"form-control\" rows=\"3\"></textarea>"
+
   const handleInputChange = (e) => {
     updateFormData({
       ...formData,
       [e.target.name]: e.target.value.trim()
     });
+
+    if(e.target.id === 'method') {
+      if(e.target.value === 'publickey') {
+        document.getElementById('div-input-pass-pub').innerHTML = INPUT_PASS_PUB
+        document.getElementById('div-input-pass-pub-label').innerText = 'Public key'
+        document.getElementById('div-input-pass-pub-genkey').style.visibility = 'visible';
```

Comment:
Use the `display` style instead so that the hidden button doesn't create extra awkward whitespace. So `display: none` when you want it hidden and `display: block` when you want it visible.

Reply:
I've had problems using display: none with react, I do have another idea though

Reply:
I've found a cleaner solution, will be in next commit